### ISSUE TITLE
Ensure Passwords are wiped from memory when dropped

### ISSUE THREAD

**Issue by simonsan** [rust-keylock/rust-keylock-android#4]
I see you use only a `String` for the passwords, maybe it would be good to adopt something like `abscissa_core::Secret` here?

https://docs.rs/abscissa_core/0.5.2/abscissa_core/struct.Secret.html

**Comment by astonbitecode** [comment]
Indeed, I didn't have the chance to deal with this in the Editors (the same stands for the `rust-keylock-ui`). The thing is that we need to have a suitable solution for Java too, because when a password reaches the Editor in plaintext (in order to be revealed to the user), it is actually sent to the Java world (transformed to a Java String via `j4rs`) in order to reach the UI. So, we need to zeroize the Java world too... 

There was a plan to use Java [ByteBuffer](http://cr.openjdk.java.net/~iris/se/11/latestSpec/api/java.base/java/nio/ByteBuffer.html) for that, but I am not sure it will be enough; we need to create a String I guess in order to actually render the password characters to the screen. I am not aware of Android components that accept `ByteBuffer`s. 

FYI, `rust-keylock-lib` (the core) uses [SecStr](https://docs.rs/secstr/0.4.0/secstr/) and [Zeroize](https://docs.rs/zeroize/1.4.3/zeroize/) to protect and wipe sensitive sensitive data (passwords, keys etc), plus keeps everything encrypted in memory until needed (like in the case discussed above - reveal a password to the user).

So, in the Editor's "Rust World", Zeroize can be used. Do you have maybe any idea on how we could pin and wipe memory in Android?
